Runtime/Interaction/Action-based/LocomotionSchemeManager.cs

```diff
@@ -0,0 +1,558 @@
+        /// Stores the "Head" <see cref="Transform"/> used with continuous movement when inputs should be relative to head orientation (usually the main camera).
+        /// </summary>
+        public Transform HeadForwardSource
+        {
+            get => headForwardSource;
+            set => headForwardSource = value;
+        }
+
+        [SerializeField]
+        [Tooltip("Stores the \"Left Hand\" Transform used with continuous movement when inputs should be relative to the left hand's orientation.")]
+        private Transform leftHandForwardSource;
+        
+        /// <summary>
+        /// Stores the "Left Hand" <see cref="Transform"/> used with continuous movement when inputs should be relative to the left hand's orientation.
+        /// </summary>
+        public Transform LeftHandForwardSource
+        {
+            get => leftHandForwardSource;
+            set => leftHandForwardSource = value;
+        }
+
+        [SerializeField]
+        [Tooltip("Stores the \"Right Hand\" Transform used with continuous movement when inputs should be relative to the right hand's orientation.")]
+        private Transform rightHandForwardSource;
+        
+        /// <summary>
+        /// Stores the "Right Hand" <see cref="Transform"/> used with continuous movement when inputs should be relative to the right hand's orientation.
+        /// </summary>
+        public Transform RightHandForwardSource
+        {
+            get => rightHandForwardSource;
+            set => rightHandForwardSource = value;
+        }
+
+        private void OnEnable()
+        {
+            SetMoveScheme(moveScheme);
+            SetTurnStyle(turnStyle);
+            SetMoveForwardSource(moveForwardSource);
+        }
+
+        private void OnDisable()
+        {
+            ClearBindingMasks();
+        }
+
+        private void SetMoveScheme(MoveSchemeType scheme)
+        {
+            switch (scheme)
+            {
+                case MoveSchemeType.Noncontinuous:
+                    SetBindingMasks(noncontinuousControlScheme);
+                    
+                    if (continuousMoveProvider != null)
+                    {
+                        continuousMoveProvider.enabled = false;
+                    }
+                    break;
+                case MoveSchemeType.Continuous:
+                    SetBindingMasks(continuousControlScheme);
+                    
+                    if (continuousMoveProvider != null)
+                    {
+                        continuousMoveProvider.enabled = true;
+                    }
+                    break;
+                default:
+                    throw new InvalidEnumArgumentException(nameof(scheme), (int)scheme, typeof(MoveSchemeType));
+            }
+        }
+
+        private void SetTurnStyle(TurnStyleType style)
+        {
+            switch (style)
+            {
+                case TurnStyleType.Snap:
+                    if (continuousTurnProvider != null)
```

Comment:
Feels like the switch consists of null checks, one liners would improve the overview.